pandas-stubs/core/indexes/datetimes.pyi

```diff
@@ -97,6 +98,7 @@ class DatetimeIndex(  # type: ignore[misc]
     def tzinfo(self) -> tzinfo | None: ...
     @property
     def dtype(self) -> np.dtype | DatetimeTZDtype: ...
+    def intersection(self, other: Index | list) -> DatetimeIndex: ...
```

Comment:
The type of `other` should be `DatetimeIndex` .

Reply:
Got it.